SCons/Node/Python.py

```diff
@@ -141,16 +141,13 @@ def get_text_contents(self):
             contents = contents + kid.get_contents().decode()
         return contents
 
-    def get_contents(self):
-        """
-        Get contents for signature calculations.
-        :return: bytes
-        """
+    def get_contents(self) -> bytes:
+        """Get contents for signature calculations."""
         text_contents = self.get_text_contents()
         try:
             return text_contents.encode()
-        except UnicodeDecodeError:
-            # Already encoded as python2 str are bytes
+        except AttributeError:
```

Comment:
Does this make sense? If the text_contents don't encode, return them anyway?
Probably best to just let an exception propagate?

Reply:
I'm okay to drop it.  The logic here was if something sent in an already ecoded (bytes) object, that will take an atrribute error - there's no encode method - so just send as is, since we wanted bytes anyway.

Reply:
Right. Then if it fails downstream, it'd hopefully die right there?

Reply:
Seemed like there was a question about this (text_contents or contents returning the wrong type) so I didn't just leave to fail here. Just as happy if we enforce it... I did add annotations on one pair of those.  So just get rid of the try block?